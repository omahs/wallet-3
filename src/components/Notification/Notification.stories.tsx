import { Meta, StoryFn } from '@storybook/react'
import React from 'react'

import Notification from './Notification'

/**
 * # Notification stories
 * These stories showcase the notification
 */
const meta: Meta<typeof Notification> = {
  title: 'components/Notification',
  component: Notification,
}

export default meta

export const Informational: StoryFn<typeof Notification.Informational> = (args) => (
  <Notification.Informational {...args} />
)
Informational.args = {
  title: 'Headline',
  description:
    'Describe the info in two sentences or less. Do not use terminal punctuation if only using one sentence. ',
  ctaLabel: 'Action 1',
  onPressCta: () => {},
  ctaLabel2: 'Action 2',
  onPressCta2: () => {},
}

export const InformationalWithoutHeadline: StoryFn<typeof Notification.Informational> = (args) => (
  <Notification.Informational {...args} />
)
InformationalWithoutHeadline.args = {
  description:
    'Describe the info in two sentences or less. Do not use terminal punctuation if only using one sentence. ',
  ctaLabel: 'Action 1',
  onPressCta: () => {},
  ctaLabel2: 'Action 2',
  onPressCta2: () => {},
}

export const InformationalWithoutActions: StoryFn<typeof Notification.Informational> = (args) => (
  <Notification.Informational {...args} />
)
InformationalWithoutActions.args = {
  description:
    'Describe the info in two sentences or less. Do not use terminal punctuation if only using one sentence. ',
}

export const InformationalWithSingleAction: StoryFn<typeof Notification.Informational> = (args) => (
  <Notification.Informational {...args} />
)
InformationalWithSingleAction.args = {
  description:
    'Describe the info in two sentences or less. Do not use terminal punctuation if only using one sentence. ',
  ctaLabel: 'Action 1',
  onPressCta: () => {},
}

export const Warn: StoryFn<typeof Notification.Warning> = (args) => (
  <Notification.Warning {...args} />
)
Warn.args = {
  title: 'Headline',
  description:
    'Describe the info in two sentences or less. Do not use terminal punctuation if only using one sentence. ',
  ctaLabel: 'Action 1',
  onPressCta: () => {},
  ctaLabel2: 'Action 2',
  onPressCta2: () => {},
}

export const WarningWithoutHeadline: StoryFn<typeof Notification.Warning> = (args) => (
  <Notification.Warning {...args} />
)
WarningWithoutHeadline.args = {
  description:
    'Describe the info in two sentences or less. Do not use terminal punctuation if only using one sentence. ',
  ctaLabel: 'Action 1',
  onPressCta: () => {},
  ctaLabel2: 'Action 2',
  onPressCta2: () => {},
}

export const WarningWithoutActions: StoryFn<typeof Notification.Warning> = (args) => (
  <Notification.Warning {...args} />
)
WarningWithoutActions.args = {
  description:
    'Describe the info in two sentences or less. Do not use terminal punctuation if only using one sentence. ',
}

export const WarningWithSingleAction: StoryFn<typeof Notification.Warning> = (args) => (
  <Notification.Warning {...args} />
)
WarningWithSingleAction.args = {
  description:
    'Describe the info in two sentences or less. Do not use terminal punctuation if only using one sentence. ',
  ctaLabel: 'Action 1',
  onPressCta: () => {},
}

export const Error: StoryFn<typeof Notification.Error> = (args) => <Notification.Error {...args} />
Error.args = {
  title: 'Headline',
  description:
    'Describe the info in two sentences or less. Do not use terminal punctuation if only using one sentence. ',
  ctaLabel: 'Action 1',
  onPressCta: () => {},
  ctaLabel2: 'Action 2',
  onPressCta2: () => {},
}

export const ErrorWithoutHeadline: StoryFn<typeof Notification.Error> = (args) => (
  <Notification.Error {...args} />
)
ErrorWithoutHeadline.args = {
  description:
    'Describe the info in two sentences or less. Do not use terminal punctuation if only using one sentence. ',
  ctaLabel: 'Action 1',
  onPressCta: () => {},
  ctaLabel2: 'Action 2',
  onPressCta2: () => {},
}

export const ErrorWithoutActions: StoryFn<typeof Notification.Error> = (args) => (
  <Notification.Error {...args} />
)
ErrorWithoutActions.args = {
  description:
    'Describe the info in two sentences or less. Do not use terminal punctuation if only using one sentence. ',
}

export const ErrorWithSingleAction: StoryFn<typeof Notification.Error> = (args) => (
  <Notification.Error {...args} />
)
ErrorWithSingleAction.args = {
  description:
    'Describe the info in two sentences or less. Do not use terminal punctuation if only using one sentence. ',
  ctaLabel: 'Action 1',
  onPressCta: () => {},
}
