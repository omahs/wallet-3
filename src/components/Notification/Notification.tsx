import React from 'react'
import { StyleSheet, Text, TextStyle, View, ViewStyle } from 'react-native'
import AttentionIcon from 'src/icons/Attention'
import Colors from 'src/styles/colors'
import fontStyles, { typeScale } from 'src/styles/fonts'
import { Spacing } from 'src/styles/styles'

interface CustomStyles {
  container: ViewStyle
  attentionIcon: Record<'color', Colors>
  ctaLabel: TextStyle
  ctaLabel2: TextStyle
}

interface Props {
  title?: string
  description: string
  styles: Partial<CustomStyles>
  ctaLabel?: string
  onPressCta?: () => void
  ctaLabel2?: string
  onPressCta2?: () => void
  testID?: string
}

type PropsWithPresetStyle = Omit<Props, 'styles'>

export function Notification({
  title,
  description,
  styles,
  ctaLabel,
  onPressCta,
  ctaLabel2,
  onPressCta2,
  testID,
}: Props) {
  return (
    <View style={[defaultStyles.container, styles?.container]} testID={testID}>
      {title && (
        <View style={defaultStyles.row}>
          <View style={defaultStyles.attentionIcon}>
            <AttentionIcon color={styles?.attentionIcon?.color ?? Colors.informationDark} />
          </View>
          <Text style={defaultStyles.titleText}>{title}</Text>
        </View>
      )}

      <View style={defaultStyles.bodyContainer}>
        <View style={[defaultStyles.row, defaultStyles.bodyRow]}>
          <View style={defaultStyles.attentionIcon}>
            {!title && (
              <AttentionIcon color={styles?.attentionIcon?.color ?? Colors.informationDark} />
            )}
          </View>
          <Text style={defaultStyles.bodyText}>{description}</Text>
        </View>

        {(ctaLabel || ctaLabel2) && (
          <View style={[defaultStyles.row, defaultStyles.ctaRow]}>
            {ctaLabel && onPressCta && (
              <Text style={[defaultStyles.ctaLabel, styles?.ctaLabel]} onPress={onPressCta}>
                {ctaLabel}
              </Text>
            )}
            {ctaLabel2 && onPressCta2 && (
              <Text style={[defaultStyles.ctaLabel2, styles?.ctaLabel2]} onPress={onPressCta2}>
                {ctaLabel2}
              </Text>
            )}
          </View>
        )}
      </View>
    </View>
  )
}

const defaultStyles = StyleSheet.create({
  container: {
    paddingHorizontal: Spacing.Regular16,
    paddingTop: Spacing.Regular16,
    paddingBottom: Spacing.Small12,
    borderRadius: Spacing.Regular16,
    width: '100%',
    gap: Spacing.Tiny4,
  },
  bodyContainer: {
    gap: Spacing.Smallest8,
  },
  row: {
    flexDirection: 'row',
    gap: Spacing.Smallest8,
    alignItems: 'center',
  },
  bodyRow: {
    alignItems: 'flex-start',
  },
  ctaRow: {
    paddingVertical: Spacing.Tiny4,
    paddingHorizontal: Spacing.Smallest8,
    justifyContent: 'flex-end',
  },
  attentionIcon: {
    padding: Spacing.Tiny4,
    width: Spacing.Thick24,
    height: Spacing.Thick24,
  },
  titleText: {
    ...fontStyles.small600,
  },
  bodyText: {
    ...typeScale.bodyXSmall,
    lineHeight: 18,
    color: Colors.dark,
    flexShrink: 1,
  },
  ctaLabel: {
    ...fontStyles.small600,
    paddingHorizontal: Spacing.Smallest8,
    paddingVertical: Spacing.Tiny4,
  },
  ctaLabel2: {
    ...fontStyles.small600,
    paddingHorizontal: Spacing.Smallest8,
    paddingVertical: Spacing.Tiny4,
  },
})

const informationalStyle: Partial<CustomStyles> = {
  container: { backgroundColor: Colors.informationFaint },
  attentionIcon: { color: Colors.informationDark },
  ctaLabel: { color: Colors.informationDark },
  ctaLabel2: { color: Colors.informationDark },
}

const warningStyle: Partial<CustomStyles> = {
  container: { backgroundColor: Colors.warningLight },
  attentionIcon: { color: Colors.warningDark },
  ctaLabel: { color: Colors.warningDark },
  ctaLabel2: { color: Colors.warningDark },
}

const errorStyle: Partial<CustomStyles> = {
  container: { backgroundColor: Colors.errorLight },
  attentionIcon: { color: Colors.errorDark },
  ctaLabel: { color: Colors.errorDark },
  ctaLabel2: { color: Colors.errorDark },
}

Notification.Informational = (props: PropsWithPresetStyle) =>
  Notification({ ...props, styles: informationalStyle })
Notification.Warning = (props: PropsWithPresetStyle) =>
  Notification({ ...props, styles: warningStyle })
Notification.Error = (props: PropsWithPresetStyle) => Notification({ ...props, styles: errorStyle })

export default Notification
