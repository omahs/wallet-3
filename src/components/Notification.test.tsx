import { fireEvent, render } from '@testing-library/react-native'
import * as React from 'react'
import Notification from 'src/components/Notification'

describe(Notification, () => {
  it('does not render CTA when onPress function is not provided', async () => {
    const { findByText } = render(
      <Notification.Informational
        title={'Title'}
        description={'Description'}
        ctaLabel={'Action 1'}
        onPressCta={() => {}}
        ctaLabel2={'Action 2'}
      />
    )

    await expect(findByText('Action 1')).resolves.toBeDefined()
    await expect(findByText('Action 2')).rejects.toThrow()
  })

  it('calls onPress function when pressed', async () => {
    const fn = jest.fn()
    const fn2 = jest.fn()
    const { findByText } = render(
      <Notification.Informational
        title={'Title'}
        description={'Description'}
        ctaLabel={'Action 1'}
        onPressCta={fn}
        ctaLabel2={'Action 2'}
        onPressCta2={fn2}
      />
    )

    fireEvent.press(await findByText('Action 1'))
    fireEvent.press(await findByText('Action 2'))

    expect(fn).toBeCalled()
    expect(fn2).toBeCalled()
  })

  it('renders correctly with headline, description and actions', () => {
    const tree = render(
      <Notification.Informational
        title={'Title'}
        description={'Description'}
        ctaLabel={'Action 1'}
        onPressCta={() => {}}
        ctaLabel2={'Action 2'}
        onPressCta2={() => {}}
      />
    )
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with description and actions', () => {
    const tree = render(
      <Notification.Informational
        description={'Description'}
        ctaLabel={'Action 1'}
        onPressCta={() => {}}
        ctaLabel2={'Action 2'}
        onPressCta2={() => {}}
      />
    )
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with description', () => {
    const tree = render(<Notification.Informational description={'Description'} />)
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with headline and description', () => {
    const tree = render(<Notification.Informational title={'Title'} description={'Description'} />)
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly Warning', () => {
    const tree = render(
      <Notification.Warning
        title={'Title'}
        description={'Description'}
        ctaLabel={'Action 1'}
        onPressCta={() => {}}
        ctaLabel2={'Action 2'}
        onPressCta2={() => {}}
      />
    )
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly Error', () => {
    const tree = render(
      <Notification.Error
        title={'Title'}
        description={'Description'}
        ctaLabel={'Action 1'}
        onPressCta={() => {}}
        ctaLabel2={'Action 2'}
        onPressCta2={() => {}}
      />
    )
    expect(tree).toMatchSnapshot()
  })
})
