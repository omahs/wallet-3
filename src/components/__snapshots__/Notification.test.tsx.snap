// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification renders correctly Error 1`] = `
<View
  style={
    [
      {
        "borderRadius": 16,
        "gap": 4,
        "paddingBottom": 12,
        "paddingHorizontal": 16,
        "paddingTop": 16,
        "width": "100%",
      },
      {
        "backgroundColor": "#FBF2F0",
      },
    ]
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 8,
      }
    }
  >
    <View
      style={
        {
          "height": 24,
          "padding": 4,
          "width": 24,
        }
      }
    >
      <svg
        fill="none"
        height={16}
        width={16}
      >
        <path
          d="M7.2 10.4h1.6V12H7.2v-1.6Zm0-6.4h1.6v4.8H7.2V4Zm.792-4C3.576 0 0 3.584 0 8s3.576 8 7.992 8C12.416 16 16 12.416 16 8s-3.584-8-8.008-8ZM8 14.4A6.398 6.398 0 0 1 1.6 8c0-3.536 2.864-6.4 6.4-6.4 3.536 0 6.4 2.864 6.4 6.4 0 3.536-2.864 6.4-6.4 6.4Z"
          fill="#C93717"
        />
      </svg>
    </View>
    <Text
      style={
        {
          "color": "#2E3338",
          "fontFamily": "Inter-SemiBold",
          "fontSize": 14,
          "lineHeight": 18,
        }
      }
    >
      Title
    </Text>
  </View>
  <View
    style={
      {
        "gap": 8,
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 8,
          },
          {
            "alignItems": "flex-start",
          },
        ]
      }
    >
      <View
        style={
          {
            "height": 24,
            "padding": 4,
            "width": 24,
          }
        }
      />
      <Text
        style={
          {
            "color": "#2E3338",
            "flexShrink": 1,
            "fontFamily": "Inter-Regular",
            "fontSize": 12,
            "letterSpacing": 0.12,
            "lineHeight": 18,
          }
        }
      >
        Description
      </Text>
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 8,
          },
          {
            "justifyContent": "flex-end",
            "paddingHorizontal": 8,
            "paddingVertical": 4,
          },
        ]
      }
    >
      <Text
        onPress={[MockFunction]}
        style={
          [
            {
              "color": "#2E3338",
              "fontFamily": "Inter-SemiBold",
              "fontSize": 14,
              "lineHeight": 18,
              "paddingHorizontal": 8,
              "paddingVertical": 4,
            },
            {
              "color": "#C93717",
            },
          ]
        }
      >
        Action 1
      </Text>
      <Text
        onPress={[MockFunction]}
        style={
          [
            {
              "color": "#2E3338",
              "fontFamily": "Inter-SemiBold",
              "fontSize": 14,
              "lineHeight": 18,
              "paddingHorizontal": 8,
              "paddingVertical": 4,
            },
            {
              "color": "#C93717",
            },
          ]
        }
      >
        Action 2
      </Text>
    </View>
  </View>
</View>
`;

exports[`Notification renders correctly Warning 1`] = `
<View
  style={
    [
      {
        "borderRadius": 16,
        "gap": 4,
        "paddingBottom": 12,
        "paddingHorizontal": 16,
        "paddingTop": 16,
        "width": "100%",
      },
      {
        "backgroundColor": "#FFF9EA",
      },
    ]
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 8,
      }
    }
  >
    <View
      style={
        {
          "height": 24,
          "padding": 4,
          "width": 24,
        }
      }
    >
      <svg
        fill="none"
        height={16}
        width={16}
      >
        <path
          d="M7.2 10.4h1.6V12H7.2v-1.6Zm0-6.4h1.6v4.8H7.2V4Zm.792-4C3.576 0 0 3.584 0 8s3.576 8 7.992 8C12.416 16 16 12.416 16 8s-3.584-8-8.008-8ZM8 14.4A6.398 6.398 0 0 1 1.6 8c0-3.536 2.864-6.4 6.4-6.4 3.536 0 6.4 2.864 6.4 6.4 0 3.536-2.864 6.4-6.4 6.4Z"
          fill="#9C6E00"
        />
      </svg>
    </View>
    <Text
      style={
        {
          "color": "#2E3338",
          "fontFamily": "Inter-SemiBold",
          "fontSize": 14,
          "lineHeight": 18,
        }
      }
    >
      Title
    </Text>
  </View>
  <View
    style={
      {
        "gap": 8,
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 8,
          },
          {
            "alignItems": "flex-start",
          },
        ]
      }
    >
      <View
        style={
          {
            "height": 24,
            "padding": 4,
            "width": 24,
          }
        }
      />
      <Text
        style={
          {
            "color": "#2E3338",
            "flexShrink": 1,
            "fontFamily": "Inter-Regular",
            "fontSize": 12,
            "letterSpacing": 0.12,
            "lineHeight": 18,
          }
        }
      >
        Description
      </Text>
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 8,
          },
          {
            "justifyContent": "flex-end",
            "paddingHorizontal": 8,
            "paddingVertical": 4,
          },
        ]
      }
    >
      <Text
        onPress={[MockFunction]}
        style={
          [
            {
              "color": "#2E3338",
              "fontFamily": "Inter-SemiBold",
              "fontSize": 14,
              "lineHeight": 18,
              "paddingHorizontal": 8,
              "paddingVertical": 4,
            },
            {
              "color": "#9C6E00",
            },
          ]
        }
      >
        Action 1
      </Text>
      <Text
        onPress={[MockFunction]}
        style={
          [
            {
              "color": "#2E3338",
              "fontFamily": "Inter-SemiBold",
              "fontSize": 14,
              "lineHeight": 18,
              "paddingHorizontal": 8,
              "paddingVertical": 4,
            },
            {
              "color": "#9C6E00",
            },
          ]
        }
      >
        Action 2
      </Text>
    </View>
  </View>
</View>
`;

exports[`Notification renders correctly with description 1`] = `
<View
  style={
    [
      {
        "borderRadius": 16,
        "gap": 4,
        "paddingBottom": 12,
        "paddingHorizontal": 16,
        "paddingTop": 16,
        "width": "100%",
      },
      {
        "backgroundColor": "#E8F8FF",
      },
    ]
  }
>
  <View
    style={
      {
        "gap": 8,
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 8,
          },
          {
            "alignItems": "flex-start",
          },
        ]
      }
    >
      <View
        style={
          {
            "height": 24,
            "padding": 4,
            "width": 24,
          }
        }
      >
        <svg
          fill="none"
          height={16}
          width={16}
        >
          <path
            d="M7.2 10.4h1.6V12H7.2v-1.6Zm0-6.4h1.6v4.8H7.2V4Zm.792-4C3.576 0 0 3.584 0 8s3.576 8 7.992 8C12.416 16 16 12.416 16 8s-3.584-8-8.008-8ZM8 14.4A6.398 6.398 0 0 1 1.6 8c0-3.536 2.864-6.4 6.4-6.4 3.536 0 6.4 2.864 6.4 6.4 0 3.536-2.864 6.4-6.4 6.4Z"
            fill="#0768AE"
          />
        </svg>
      </View>
      <Text
        style={
          {
            "color": "#2E3338",
            "flexShrink": 1,
            "fontFamily": "Inter-Regular",
            "fontSize": 12,
            "letterSpacing": 0.12,
            "lineHeight": 18,
          }
        }
      >
        Description
      </Text>
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 8,
          },
          {
            "justifyContent": "flex-end",
            "paddingHorizontal": 8,
            "paddingVertical": 4,
          },
        ]
      }
    />
  </View>
</View>
`;

exports[`Notification renders correctly with description and actions 1`] = `
<View
  style={
    [
      {
        "borderRadius": 16,
        "gap": 4,
        "paddingBottom": 12,
        "paddingHorizontal": 16,
        "paddingTop": 16,
        "width": "100%",
      },
      {
        "backgroundColor": "#E8F8FF",
      },
    ]
  }
>
  <View
    style={
      {
        "gap": 8,
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 8,
          },
          {
            "alignItems": "flex-start",
          },
        ]
      }
    >
      <View
        style={
          {
            "height": 24,
            "padding": 4,
            "width": 24,
          }
        }
      >
        <svg
          fill="none"
          height={16}
          width={16}
        >
          <path
            d="M7.2 10.4h1.6V12H7.2v-1.6Zm0-6.4h1.6v4.8H7.2V4Zm.792-4C3.576 0 0 3.584 0 8s3.576 8 7.992 8C12.416 16 16 12.416 16 8s-3.584-8-8.008-8ZM8 14.4A6.398 6.398 0 0 1 1.6 8c0-3.536 2.864-6.4 6.4-6.4 3.536 0 6.4 2.864 6.4 6.4 0 3.536-2.864 6.4-6.4 6.4Z"
            fill="#0768AE"
          />
        </svg>
      </View>
      <Text
        style={
          {
            "color": "#2E3338",
            "flexShrink": 1,
            "fontFamily": "Inter-Regular",
            "fontSize": 12,
            "letterSpacing": 0.12,
            "lineHeight": 18,
          }
        }
      >
        Description
      </Text>
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 8,
          },
          {
            "justifyContent": "flex-end",
            "paddingHorizontal": 8,
            "paddingVertical": 4,
          },
        ]
      }
    >
      <Text
        onPress={[MockFunction]}
        style={
          [
            {
              "color": "#2E3338",
              "fontFamily": "Inter-SemiBold",
              "fontSize": 14,
              "lineHeight": 18,
              "paddingHorizontal": 8,
              "paddingVertical": 4,
            },
            {
              "color": "#0768AE",
            },
          ]
        }
      >
        Action 1
      </Text>
      <Text
        onPress={[MockFunction]}
        style={
          [
            {
              "color": "#2E3338",
              "fontFamily": "Inter-SemiBold",
              "fontSize": 14,
              "lineHeight": 18,
              "paddingHorizontal": 8,
              "paddingVertical": 4,
            },
            {
              "color": "#0768AE",
            },
          ]
        }
      >
        Action 2
      </Text>
    </View>
  </View>
</View>
`;

exports[`Notification renders correctly with headline and description 1`] = `
<View
  style={
    [
      {
        "borderRadius": 16,
        "gap": 4,
        "paddingBottom": 12,
        "paddingHorizontal": 16,
        "paddingTop": 16,
        "width": "100%",
      },
      {
        "backgroundColor": "#E8F8FF",
      },
    ]
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 8,
      }
    }
  >
    <View
      style={
        {
          "height": 24,
          "padding": 4,
          "width": 24,
        }
      }
    >
      <svg
        fill="none"
        height={16}
        width={16}
      >
        <path
          d="M7.2 10.4h1.6V12H7.2v-1.6Zm0-6.4h1.6v4.8H7.2V4Zm.792-4C3.576 0 0 3.584 0 8s3.576 8 7.992 8C12.416 16 16 12.416 16 8s-3.584-8-8.008-8ZM8 14.4A6.398 6.398 0 0 1 1.6 8c0-3.536 2.864-6.4 6.4-6.4 3.536 0 6.4 2.864 6.4 6.4 0 3.536-2.864 6.4-6.4 6.4Z"
          fill="#0768AE"
        />
      </svg>
    </View>
    <Text
      style={
        {
          "color": "#2E3338",
          "fontFamily": "Inter-SemiBold",
          "fontSize": 14,
          "lineHeight": 18,
        }
      }
    >
      Title
    </Text>
  </View>
  <View
    style={
      {
        "gap": 8,
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 8,
          },
          {
            "alignItems": "flex-start",
          },
        ]
      }
    >
      <View
        style={
          {
            "height": 24,
            "padding": 4,
            "width": 24,
          }
        }
      />
      <Text
        style={
          {
            "color": "#2E3338",
            "flexShrink": 1,
            "fontFamily": "Inter-Regular",
            "fontSize": 12,
            "letterSpacing": 0.12,
            "lineHeight": 18,
          }
        }
      >
        Description
      </Text>
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 8,
          },
          {
            "justifyContent": "flex-end",
            "paddingHorizontal": 8,
            "paddingVertical": 4,
          },
        ]
      }
    />
  </View>
</View>
`;

exports[`Notification renders correctly with headline, description and actions 1`] = `
<View
  style={
    [
      {
        "borderRadius": 16,
        "gap": 4,
        "paddingBottom": 12,
        "paddingHorizontal": 16,
        "paddingTop": 16,
        "width": "100%",
      },
      {
        "backgroundColor": "#E8F8FF",
      },
    ]
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 8,
      }
    }
  >
    <View
      style={
        {
          "height": 24,
          "padding": 4,
          "width": 24,
        }
      }
    >
      <svg
        fill="none"
        height={16}
        width={16}
      >
        <path
          d="M7.2 10.4h1.6V12H7.2v-1.6Zm0-6.4h1.6v4.8H7.2V4Zm.792-4C3.576 0 0 3.584 0 8s3.576 8 7.992 8C12.416 16 16 12.416 16 8s-3.584-8-8.008-8ZM8 14.4A6.398 6.398 0 0 1 1.6 8c0-3.536 2.864-6.4 6.4-6.4 3.536 0 6.4 2.864 6.4 6.4 0 3.536-2.864 6.4-6.4 6.4Z"
          fill="#0768AE"
        />
      </svg>
    </View>
    <Text
      style={
        {
          "color": "#2E3338",
          "fontFamily": "Inter-SemiBold",
          "fontSize": 14,
          "lineHeight": 18,
        }
      }
    >
      Title
    </Text>
  </View>
  <View
    style={
      {
        "gap": 8,
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 8,
          },
          {
            "alignItems": "flex-start",
          },
        ]
      }
    >
      <View
        style={
          {
            "height": 24,
            "padding": 4,
            "width": 24,
          }
        }
      />
      <Text
        style={
          {
            "color": "#2E3338",
            "flexShrink": 1,
            "fontFamily": "Inter-Regular",
            "fontSize": 12,
            "letterSpacing": 0.12,
            "lineHeight": 18,
          }
        }
      >
        Description
      </Text>
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 8,
          },
          {
            "justifyContent": "flex-end",
            "paddingHorizontal": 8,
            "paddingVertical": 4,
          },
        ]
      }
    >
      <Text
        onPress={[MockFunction]}
        style={
          [
            {
              "color": "#2E3338",
              "fontFamily": "Inter-SemiBold",
              "fontSize": 14,
              "lineHeight": 18,
              "paddingHorizontal": 8,
              "paddingVertical": 4,
            },
            {
              "color": "#0768AE",
            },
          ]
        }
      >
        Action 1
      </Text>
      <Text
        onPress={[MockFunction]}
        style={
          [
            {
              "color": "#2E3338",
              "fontFamily": "Inter-SemiBold",
              "fontSize": 14,
              "lineHeight": 18,
              "paddingHorizontal": 8,
              "paddingVertical": 4,
            },
            {
              "color": "#0768AE",
            },
          ]
        }
      >
        Action 2
      </Text>
    </View>
  </View>
</View>
`;
